ansible/roles/postgresql/tasks/main.yml

```diff
@@ -74,49 +87,46 @@
   become: true
   service: name=postgresql state=started
   tags: after-reboot
+  register: pg_started
 
-- name: Create PostgreSQL users
-  become: yes
-  become_user: postgres
-  vars:
-    ansible_ssh_pipelining: true
-  postgresql_user:
-    name: "{{ localsettings.PG_DATABASE_USER }}"
-    password: "{{ localsettings.PG_DATABASE_PASSWORD }}"
-    role_attr_flags: CREATEDB
-    state: present
-    port: "{{ postgresql_port }}"
-  when: not is_pg_standby|default(False)
-  tags:
-    - postgres_users
+- name: restart postgresql if necessary
+  become: true
+  service: name=postgresql state=restarted
+  when: pg_config_changed.changed and not pg_started.changed
```

Comment:
This feels very hacky but a restart is required after setting `max_replication_slots` before any replication slots can be created.

Reply:
Assuming you're saying that this is hacky because it restarts postgres on _any_ configuration change, not just a change in `max_replication_slots`. If that's correct, would it be possible to read `max_replication_slots` from the config file at the beginning of the ansible run and only restart if it changed?

Reply:
Switched to use `flush_handlers` (4499137)